### PR TITLE
Remove the detection of Jet/Centos7 nodes

### LINKED ISSUE #919: Build failing on some front nodes of Jet
Upp build is failing on some front nodes of Jet because of failure to locate some modules.

### PR DESCRIPTION
**Pull request by InnocentSouopgui-NOAA**
- After the full migration of Jet to Centos8, it is no longer necessary to detect centos7 frontend; doing so results in the failure to build.
- Remove the detection of Jet/Centos7 nodes and removes the associated module file jet_c.lua
- This resolves issue #919
- Refs NOAA-EMC/global-workflow#2377

### CODE REVIEW

File: tests/detect_machine.sh [hunk 1]
```diff
@@ -30,8 +30,7 @@ case $(hostname -f) in
 
   s4-submit.ssec.wisc.edu) MACHINE_ID=s4 ;; ### s4
 
-  fe[1-4]) MACHINE_ID=jet_c ;; ### jet1-4
-  fe[5-8]) MACHINE_ID=jet ;; ### jet1-8
+  fe[1-8]) MACHINE_ID=jet ;; ### jet1-8
```
Comment (WenMeng-NOAA) [inline tests/detect_machine.sh:33]: @InnocentSouopgui-NOAA It seems to me the fe2 is still CentOS7 node. Do you know when it will be become Rocky8 node?

Comment (FernandoAndrade-NOAA) [reply tests/detect_machine.sh:33]: @WenMeng-NOAA I believe the final transition date for Jet is 4/9, next Tuesday.

Comment (InnocentSouopgui-NOAA) [reply tests/detect_machine.sh:33]: That is correct, I noticed that as well, fe2 is still CentOS7. The team in charge of migrating to Rocky8 announced last week that they migrated everything (frontend nodes and partitions nodes) to Rocky8;
I don't know when they will migrate fe2 to Rocky8

Comment (WenMeng-NOAA) [reply tests/detect_machine.sh:33]: Okay, let's assume the jet final transition includes fe2.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Can you test UPP building on jet with this PR?

**Comment by FernandoAndrade-NOAA** [comment]
FYI I've made an additional change to remove Hera's hfe12, it's a restricted node on centos following the recent maintenance.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA Can you test UPP building on jet with this PR?

Building completed on Jet, I did not see any issues in the output. This was on fe4.

**Comment by WenMeng-NOAA** [comment]
@InnocentSouopgui-NOAA Can you sync your branch with the latest UPP develop branch for our merging process?

**Comment by InnocentSouopgui-NOAA** [comment]
> @InnocentSouopgui-NOAA Can you sync your branch with the latest UPP develop branch for our merging process?

Done.

**Comment by WenMeng-NOAA** [comment]
@InnocentSouopgui-NOAA Thanks for contributing UPP updates for Jet Rocky8 transition.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs were completed on WCOSS2. No baseline update is needed with this PR.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Please let me know if you have more comments on this PR. I think testing on WCOSS2 should be adequate.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA Please let me know if you have more comments on this PR. I think testing on WCOSS2 should be adequate.

I will go ahead and run a quick compile on Hera to ensure it's still detecting 10-11 correctly, that should be enough given nothing else was updated.

**Comment by FernandoAndrade-NOAA** [comment]
Looks good to go from my side, thank you.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**